util/types/datum.go

```diff
@@ -1124,6 +1174,26 @@ func (d *Datum) convertToMysqlSet(sc *variable.StatementContext, target *FieldTy
 	return ret, nil
 }
 
+func (d *Datum) convertToMysqlJson(sc *variable.StatementContext, target *FieldType) (Datum, error) {
+	var (
+		s   string
+		j   Json = CreateJson(nil)
+		ret Datum
+		err error = nil
```

Comment:
No need to declare err here. According to your function, use err := j.ParseFromString(s), keep Go-ish.